tests/bluetooth/bsim_bt/bsim_test_per_adv/src/per_adv_syncer.c

```diff
@@ -134,9 +135,37 @@ static void term_cb(struct bt_le_per_adv_sync *sync,
 	SET_FLAG(flag_per_adv_sync_lost);
 }
 
+static void recv_cb(struct bt_le_per_adv_sync *recv_sync,
+		    const struct bt_le_per_adv_sync_recv_info *info,
+		    struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	uint8_t buf_data_len;
+
+	if (TEST_FLAG(flag_per_adv_recv)) {
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	printk("PER_ADV_SYNC[%u]: [DEVICE]: %s advertisment received\n",
+	       bt_le_per_adv_sync_get_index(recv_sync), le_addr);
+
+	while (buf->len > 0) {
+		buf_data_len = (uint8_t)net_buf_simple_pull_le16(buf);
+		if (buf->data[0]-1 != ARRAY_SIZE(mfg_data) ||
```

Comment:
```suggestion
		if (buf->data[0] - 1 != sizeof(mfg_data) ||
```

The length field is in octets, and not number of elements. Of course when using `uint8_t` that's the same, but in case that `mfg_data` would every change type from `uint8_t`, using `ARRAY_SIZE` wouldn't work